contract/models/abstract_contract.py

```diff
@@ -48,6 +48,20 @@ class ContractAbstractContract(models.AbstractModel):
         help="Mark this check if you want to control recurrrence at line level instead"
         " of all together for the whole contract.",
     )
+    generation_type = fields.Selection(
+        string="Generation Type",
+        selection=lambda self: self._get_generation_type_selection(),
+        default=lambda self: self._get_default_generation_type(),
+        help="Choose the document that will be automatically generated by cron.",
+    )
+
+    @api.model
+    def _get_generation_type_selection(self):
```

Comment:
Please use conventions for method naming (https://www.odoo.com/documentation/15.0/es/developer/misc/other/guidelines.html#symbols-and-conventions): `_selection_generation_type` and `_default_generation_type`.

Reply:
Arf didn't know Odoo has done such requirements :smile: But great!